irohad/ametsuchi/impl/redis_block_query.cpp

```diff
@@ -170,6 +170,22 @@ namespace iroha {
           });
     }
 
+    rxcpp::observable<boost::optional<model::Transaction>>
+    RedisBlockQuery::getTransactions(
+        const std::vector<iroha::hash256_t> &tx_hashes) {
+      return rxcpp::observable<>::create<boost::optional<model::Transaction>>(
+          [this, tx_hashes](auto subscriber) {
+            std::for_each(tx_hashes.begin(),
+                          tx_hashes.end(),
+                          [ that = this, &subscriber ](auto tx_hash) {
+                            auto tx =
+                                that->getTxByHashSync(tx_hash.to_string());
+                            subscriber.on_next(tx);
```

Comment:
is it necessary to copy `tx_hash` inside this lambda or it can be passed by ref?
Also, is it necessary to copy tx twice:
 1. `auto tx = ... `
 2. `subscriber.on_next(tx)` 

Maybe it is possible to omit one copy?

Reply:
> is it necessary to copy tx_hash inside this lambda or it can be passed by ref?

It should be copied because test fails. The reason is maybe lazy-evaluation.

And, I am afraid that capturing `this` in this and other methods. Can we guarantee that the object is alive until begin subscribed?

> is it necessary to copy tx twice:

Thanks, that's right.